### PR TITLE
feat: setup release process for CLI

### PR DESCRIPTION
**Pull request by jonaslagoni**
## Description
This PR setup the release process for the CLI so it is released alongside Modelina with the same versions. Also, each GitHub release will have the proper assets as the CLI.

This means that going forwards we will have a new published package called `@asyncapi/modelina-cli` published to npm, Homebrew, Chocolatey, and GitHub releases will contain `macos`, `win32`, `deb`, `tarballs` files for the CLI.

Questions:
1. is there anything we need to do to release on Homebrew?
1. is there anything we need to do to release on Chocolatey?

TODO before the release of the CLI:

- [x] Add CHOCOLATEY_API_KEY as secret
- [x] Add GH_TOKEN_BOT_EVE as secret

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,13 +123,18 @@
     "test:watch": "jest --watch",
     "docs": "npm run docs:api",
     "docs:api": "typedoc src/index.ts --out ./modelina-website/public/apidocs/generated --name Modelina",
-    "lint": "eslint --max-warnings 0 --config .eslintrc .",
+    "lint": "npm run lint:modelina && npm run lint:cli",
+    "lint:modelina": "eslint --max-warnings 0 --config .eslintrc .",
+    "lint:cli": "cd modelina-cli && npm run lint",
     "lint:fix": "eslint --max-warnings 0 --config .eslintrc . --fix",
-    "release": "semantic-release",
```
Comment (jonaslagoni) [inline package.json:126]: Left over script that are no longer used.

**Comment by derberg** [comment]
`CHOCOLATEY_API_KEY` added, and the eve bot token you already have access to as it is generic org token

File: .github/workflows/deploy/chocolatey/asyncapi-modelina.nuspec [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
+<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
+  <metadata>
+    <id>asyncapi</id>
```
Comment (Shurtu-gal) [inline .github/workflows/deploy/chocolatey/asyncapi-modelina.nuspec:5]: You will have to change the id to something else as this won't work. Maybe modelina can be the id.

File: .github/workflows/deploy/chocolatey/tools/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  checksum      = '{{checksum}}'
+  checksumType  = 'sha256' #default is md5, can also be sha1, sha256 or sha512
+  checksum64    = '{{checksum64}}'
+  checksumType64= 'sha256' #default is checksumType
+
+  validExitCodes= @(0, 3010, 1641)
+  silentArgs   = '/S'           # NSIS
+}
+
+Install-ChocolateyPackage @packageArgs # https://docs.chocolatey.org/en-us/create/functions/install-chocolateypackage
```
Comment (Shurtu-gal) [inline .github/workflows/deploy/chocolatey/tools/chocolateyinstall.ps1:26]: Would suggest trying to build a nupkg once and see if installation is working.

File: .github/workflows/deploy/chocolatey/asyncapi-modelina.nuspec (hunk 2 above)
Comment (jonaslagoni) [reply .github/workflows/deploy/chocolatey/asyncapi-modelina.nuspec:5]: Changing 👍

File: .github/workflows/deploy/chocolatey/tools/chocolateyinstall.ps1 (hunk 3 above)
Comment (jonaslagoni) [reply .github/workflows/deploy/chocolatey/tools/chocolateyinstall.ps1:26]: Will do 👍

**Comment by devilkiller-ag** [comment]
Hi @jonaslagoni, I will review this PR today. Sorry for being late.

**Comment by jonaslagoni** [comment]
@devilkiller-ag no rush, you can also review it in hindsight and I can still apply the changes 🙂

**Review by devilkiller-ag — APPROVED** [review]
Looks good from my side! However, I haven't checked if all the installation methods are working properly or not.

**Comment by jonaslagoni** [comment]
@devilkiller-ag thanks for checking it out ✌️ There is probably some problems we have to smooth out anyway after the merge 😄

**Pull request merged**